stempy/electron_mpi.cpp

```diff
@@ -88,8 +87,7 @@ void sendPartialMap(std::vector<Events>& events)
            MPI_BYTE, 0, 0, MPI_COMM_WORLD);
 }
 
-void gatherEvents(int worldSize, int rank,
-                 std::vector<Events>& events)
+void gatherEvents(int worldSize, int rank, Events& events)
```

Comment:
@cjh1 For the MPI functions, I noticed that a `std::vector<Events>&` was being used, where the outer vector represented the scan position.

Now, an extra vector was added to the `Events` object, and the outer vector represents the scan position for it. So I am hoping that just removing the `std::vector<>` from around the `Events` will be enough. It compiles at least. Can you try it out and see if it works?